syft/pkg/npm_package_json_metadata.go

```diff
@@ -1,11 +1,32 @@
 package pkg
 
+import (
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/syft/linux"
+)
+
+var _ urlIdentifier = (*NpmPackageJSONMetadata)(nil)
+
 // NpmPackageJSONMetadata holds extra information that is used in pkg.Package
 type NpmPackageJSONMetadata struct {
+	Name        string   `mapstructure:"name" json:"name"`
+	Version     string   `mapstructure:"version" json:"version"`
 	Files       []string `mapstructure:"files" json:"files,omitempty"`
 	Author      string   `mapstructure:"author" json:"author"`
 	Licenses    []string `mapstructure:"licenses" json:"licenses"`
 	Homepage    string   `mapstructure:"homepage" json:"homepage"`
 	Description string   `mapstructure:"description" json:"description"`
 	URL         string   `mapstructure:"url" json:"url"`
 }
+
+// PackageURL returns the PURL for the specific NPM package (see https://github.com/package-url/purl-spec)
+func (p NpmPackageJSONMetadata) PackageURL(_ *linux.Release) string {
+	return packageurl.NewPackageURL(
+		packageurl.TypeNPM,
+		"",
+		p.Name,
+		p.Version,
+		nil,
+		"",
+	).ToString()
```

Comment:
NIT: go's https://pkg.go.dev/fmt#Stringer uses `String() string` as signature.

Reply:
So for example 
```golang
purl := packageurl.NewPackageURL(
 		packageurl.TypeNPM,
 		"",
 		p.Name,
 		p.Version,
 		nil,
 		"",
 	)

fmt.Printf("purl: %s", purl) // Printf will look for purl.String()
```

Reply:
nvm there is an alternative: https://github.com/package-url/packageurl-go/blob/master/packageurl.go#L189

Reply:
it seems like `p.ToString()` and `p.String()` are really the same call on this object (not certain why they didn't just provide `String` and leave off `ToString`)